### DISCUSSION TITLE
After using `g?` to show mappings, how do I hide the mappings without needing to toggle NvimTree twice?

### DISCUSSION THREAD

**Discussion by villapx-path** [nvim-tree/nvim-tree.lua#2777] [Q&A] [answered]
As the title states, when NvimTree is open, after I type the `g?` mapping to show all keymaps, how do I "hide" the mappings to see my file tree again without needing to close NvimTree then re-open it?

If it makes any difference, I have it configured as a floating window:

```lua
  {
    "nvim-tree/nvim-tree.lua",
    version = "*",
    lazy = false,
    dependencies = { "nvim-tree/nvim-web-devicons", },
    config = function()
      require("nvim-tree").setup({
        view = {
          centralize_selection = true,
          float = {
            enable = true,
            open_win_config = {
              width = 50,
              height = 50,
            },
          },
        },
      })
    end,
  },
```

Thanks in advance!

**Comment by gegoune** [comment]
Try `q`. :)

**Reply by villapx-path** [reply]
@gegoune That closes NvimTree completely, not just the 'mappings'; I'm hoping to close the mappings but still have NvimTree open

**Reply by gegoune** [reply]
It closes help window only for both minimal and my personal config. Might have something to do with floating window, which I don't use. Can you try without floating window configuration?

**Reply by villapx-path** [reply]
@gegoune Yep, looks like `q` does work without using a floating NvimTree window.

I'm hoping there's a way I can use floating and have `q` behave that same way

**Reply by gegoune** [reply, marked as answer]
It does as well actually, I just checked and it works as you expect (with floating window configured as per your report). This points to some other culprit in your configuration.

**Reply by villapx-path** [reply]
@gegoune Hmm, interesting. Thank you for your help on this!

I'll start from scratch in my [config](https://github.com/villapx/home/blob/neovim/.config/nvim/init.lua), and slowly add items until I figure out what exactly is the culprit.

**Reply by gegoune** [reply]
Check what `:map q` returns with that window opened.

**Reply by villapx-path** [reply]
Hmm, this seems like what it should be:
```
n  q           *@<Lua 166: ~/.local/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/help.lua:152>
                 nvim-tree: exit help
```

Comparing this to the output when the 'mappings' window is not open:
```
n  q           *@<Lua 232: ~/.local/share/nvim/lazy/nvim-tree.lua/lua/nvim-tree/api.lua:50>
                 nvim-tree: Close
```

**Reply by villapx-path** [reply]
@gegoune Hmm, even with everything else besides lazy.nvim and nvim-tree stripped out of my config, I still observe this behavior:

```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not (vim.uv or vim.loop).fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    "--branch=stable", -- latest stable release
    lazypath,
  })
end
vim.opt.rtp:prepend(lazypath)

require("lazy").setup({
  {
    "nvim-tree/nvim-tree.lua",
    version = "*",
    lazy = false,
    dependencies = { "nvim-tree/nvim-web-devicons", },
    config = function()
      require("nvim-tree").setup({
        view = {
          centralize_selection = true,
          float = {
            enable = true,
            open_win_config = {
              width = 50,
              height = 50,
            },
          },
        },
      })
    end,
  },
})

vim.g.mapleader = " "

local nvimtree = require("nvim-tree.api")
vim.keymap.set("n", "<Leader>t", function() nvimtree.tree.toggle(); end)
vim.keymap.set("n", "<Leader>T", function() nvimtree.tree.find_file({ open = true, focus = true, }); end)
```

Do you suspect that either lazy.nvim, `vim.g.loaded_netrw = 1` or `vim.g.loaded_netrwPlugin = 1` are the culprit?

**Reply by gegoune** [reply]
Can you try without lazy.nvim?

**Reply by villapx-path** [reply]
Sorry I went so much time go by without responding, work got very busy and I forgot!

This issue has gone away by now; everything has been updated since May, and I'm not sure if updating Neovim, Lazy or nvim-tree is what did the trick here. Apologies that I couldn't find a specific culprit, but I'm glad it's working as expected now 😁

**Comment by alex-courtis** [comment]
This prompted me into action to fix a minor annoyance: help now closes on `<Esc>` as well as the mappings to open help.

#2909
